### PR TITLE
`Communication`: Add Post Action Bar to Thread View

### PR DESCRIPTION
**Pull request by julian-wls**
### Problem Description

The Post Actions are not as accessible through the bottom sheet dialog. The iOS app has a post action bar placed below the parent post in the thread view.

### Changes

This PR introduces the post action bar for Android, allowing users to easily access important post actions. 
The post action bar has been placed between the standalone post and the answer posts in MetisThreadUI. Post colors have been moved to a common object in the ui package.

### Steps for testing

1. Go to a chat.
2. Click on a post to navigate to the thread UI.
3. Verify that the answer post count is correct.
4. Verify that the following post actions are available:
    - deleting for authors and moderators
    - editing for authors
    - pining for moderators in channels, for group creators in groups
    - copying for everyone

### Screenshots

postActionBar (1)

### CODE REVIEW

**Review by FelberMartin — CHANGES_REQUESTED** [review]
I started with the review but noticed that there are also other changes included (eg from the introduction of the post pinning feature). Additionally there are some conflicts that I probably introduced with the latest merges. 
As soon as these are resolved, please re-request my review :D

I also include some first code comments that I noticed, you can already tackle them if you want, or wait with resolving for my full review after the conflicts are solved :)

File: feature/metis/conversation/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ui/post/post_actions/PostActionBar.kt [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                    ImageVector.vectorResource(
+                        R.drawable.unpin
+                    ), null, tint = MaterialTheme.colorScheme.primary
+                ) else Icon(ImageVector.vectorResource(R.drawable.pin), null, tint = MaterialTheme.colorScheme.primary)
+            }
+        }
+
+        if (postActions.requestEditPost != null || postActions.requestDeletePost != null) {
+            VerticalDivider()
+        }
```
Comment (FelberMartin) [inline feature/metis/conversation/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ui/post/post_actions/PostActionBar.kt:109]: I'm a big fan of return early, we could use this here :D
```
if (postActions.requestEditPost == null && postActions.requestDeletePost == null) {
    return
}

VerticalDivider()
```

**Review by FelberMartin — APPROVED** [review]
Everything looking good and working as expected

File: feature/metis/conversation/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ui/post/post_actions/PostActionBar.kt (hunk 1 above)
Comment (julian-wls) [reply feature/metis/conversation/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ui/post/post_actions/PostActionBar.kt:109]: True :)

**Pull request merged**